DeviceSDK/src/main/java/org/astarteplatform/devicesdk/util/UUID5.java

```diff
@@ -0,0 +1,61 @@
+package org.astarteplatform.devicesdk.util;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * This class is based on stackoverflow question, with some changes.
+ *
+ * @see <a
+ *     href="https://stackoverflow.com/questions/40230276/how-to-make-a-type-5-uuid-in-java">UUID5
+ *     Java</a>
+ */
+public class UUID5 {
+  private static final Charset UTF8 = StandardCharsets.UTF_8;
+
+  public static UUID nameUUIDFromNamespaceAndString(UUID namespace, String data) {
+    return nameUUIDFromNamespaceAndBytes(
+        namespace, Objects.requireNonNull(data, "data == null").getBytes(UTF8));
+  }
+
+  public static UUID nameUUIDFromNamespaceAndBytes(UUID namespace, byte[] data) {
```

Comment:
Missing javadoc also here.

Reply:
I also don't understand why we are not relying on function overloading rather using different method names.